### DISCUSSION TITLE
The game crashed whilst initializing game Error: java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'plasmovoice'!

### DISCUSSION THREAD

**Discussion by Saihilil** [FabricMC/fabric-api#3072] [Player support]
Не заходит в Майнкрафт с модом plazmovoice, и при чём это только на версиях мода выше 2.0.0. Подскажите пожалуйста, в чём проблема и как её можно решить.

---- Minecraft Crash Report ----
// Uh... Did I do that?

Time: 2023-05-23 20:18:52
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'plasmovoice'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:465)
	at net.minecraft.client.main.Main.method_44604(Main.java:205)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.NullPointerException: Cannot invoke "su.plo.voice.client.config.VoiceClientConfig.setConfigFile(java.io.File)" because "this.config" is null
	at su.plo.voice.client.BaseVoiceClient.loadConfig(BaseVoiceClient.java:290)
	at su.plo.voice.client.BaseVoiceClient.onInitialize(BaseVoiceClient.java:186)
	at su.plo.voice.client.ModVoiceClient.onInitializeClient(ModVoiceClient.java:133)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
	... 8 more

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:465)

-- Initialization --
Details:
	Modules: 
		ADVAPI32.dll:Расширенная библиотека API Windows 32:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		COMCTL32.dll:Библиотека элементов управления взаимодействия с пользователем:6.10 (WinBuild.160101.0800):Microsoft Corporation
		CRYPT32.dll:API32 криптографии:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTBASE.dll:Base cryptographic API DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTSP.dll:Cryptographic Service Provider API:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		DBGHELP.DLL:Windows Image Helper:10.0.19041.867 (WinBuild.160101.0800):Microsoft Corporation
		DNSAPI.dll:Динамическая библиотека API DNS-клиента:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		GDI32.dll:GDI Client DLL:10.0.19041.2130 (WinBuild.160101.0800):Microsoft Corporation
		IMM32.DLL:Multi-User Windows IMM32 API Client DLL:10.0.19041.2193 (WinBuild.160101.0800):Microsoft Corporation
		IPHLPAPI.DLL:API вспомогательного приложения IP:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		KERNEL32.DLL:Библиотека клиента Windows NT BASE API:10.0.19041.1741 (WinBuild.160101.0800):Microsoft Corporation
		KERNELBASE.dll:Библиотека клиента Windows NT BASE API:10.0.19041.1741 (WinBuild.160101.0800):Microsoft Corporation
		MpOav.dll:IOfficeAntiVirus Module:4.18.2304.8 (WinBuild.160101.0800):Microsoft Corporation
		NLAapi.dll:Network Location Awareness 2:10.0.19041.2193 (WinBuild.160101.0800):Microsoft Corporation
		NSI.dll:NSI User-mode interface DLL:10.0.19041.610 (WinBuild.160101.0800):Microsoft Corporation
		NTASN1.dll:Microsoft ASN.1 API:10.0.19041.320 (WinBuild.160101.0800):Microsoft Corporation
		Ole32.dll:Microsoft OLE для Windows:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		OleAut32.dll:OLEAUT32.DLL:10.0.19041.985 (WinBuild.160101.0800):Microsoft Corporation
		PSAPI.DLL:Process Status Helper:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		Pdh.dll:Модуль поддержки данных производительности Windows:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		RPCRT4.dll:Библиотека удаленного вызова процедур:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		SHCORE.dll:SHCORE:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		SHELL32.dll:Общая библиотека оболочки Windows:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		USER32.dll:Многопользовательская библиотека клиента USER API Windows:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		USERENV.dll:Userenv:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		VCRUNTIME140.dll:Microsoft® C Runtime Library:14.29.30139.0 built by: vcwrkspc:Microsoft Corporation
		VERSION.dll:Version Checking and File Installation Libraries:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		WINHTTP.dll:Службы HTTP Windows:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		WINMM.dll:MCI API DLL:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		WS2_32.dll:32-разрядная библиотека Windows Socket 2.0:10.0.19041.1081 (WinBuild.160101.0800):Microsoft Corporation
		WSOCK32.dll:Windows Socket 32-Bit DLL:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		Wldp.dll:Политика блокировки Windows:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		amsi.dll:Anti-Malware Scan Interface:10.0.19041.2075 (WinBuild.160101.0800):Microsoft Corporation
		antimalware_provider.dll:Kaspersky AntiMalwareProvider Component:30.587.0.880:AO Kaspersky Lab
		bcrypt.dll:Библиотека криптографических примитивов Windows:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		bcryptPrimitives.dll:Windows Cryptographic Primitives Library:10.0.19041.2486 (WinBuild.160101.0800):Microsoft Corporation
		clbcatq.dll:COM+ Configuration Catalog:2001.12.10941.16384 (WinBuild.160101.0800):Microsoft Corporation
		combase.dll:Microsoft COM для Windows:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		dbgcore.DLL:Windows Core Debugging Helpers:10.0.19041.2311 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc.DLL:Служба DHCP-клиента:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc6.DLL:Клиент DHCPv6:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		fwpuclnt.dll:API пользовательского режима FWP/IPsec:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		gdi32full.dll:GDI Client DLL:10.0.19041.2604 (WinBuild.160101.0800):Microsoft Corporation
		glfw.dll:GLFW 3.4.0 DLL:3.4.0:GLFW
		java.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		javaw.exe:OpenJDK Platform binary:17.0.3.0:Microsoft
		jemalloc.dll
		jimage.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		jli.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		jna11843490935983835910.dll:JNA native library:6.1.4:Java(TM) Native Access (JNA)
		jsvml.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		jvm.dll:OpenJDK 64-Bit server VM:17.0.3.0:Microsoft
		kernel.appcore.dll:AppModel API Host:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		lwjgl.dll
		management.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		management_ext.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		msvcp140.dll:Microsoft® C Runtime Library:14.29.30139.0 built by: vcwrkspc:Microsoft Corporation
		msvcp_win.dll:Microsoft® C Runtime Library:10.0.19041.789 (WinBuild.160101.0800):Microsoft Corporation
		msvcrt.dll:Windows NT CRT DLL:7.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		mswsock.dll:Расширение поставщика службы API Microsoft Windows Sockets 2.0:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		napinsp.dll:Поставщик оболочки совместимости для имен электронной почты:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		ncrypt.dll:Маршрутизатор Windows NCrypt:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		net.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		nio.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		ntdll.dll:Системная библиотека NT:10.0.19041.1741 (WinBuild.160101.0800):Microsoft Corporation
		pnrpnsp.dll:Поставщик пространства имен PNRP:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		profapi.dll:User Profile Basic API:10.0.19041.844 (WinBuild.160101.0800):Microsoft Corporation
		rasadhlp.dll:Remote Access AutoDial Helper:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		rsaenh.dll:Microsoft Enhanced Cryptographic Provider:10.0.19041.320 (WinBuild.160101.0800):Microsoft Corporation
		sechost.dll:Host for SCM/SDDL/LSA Lookup APIs:10.0.19041.320 (WinBuild.160101.0800):Microsoft Corporation
		shlwapi.dll:Библиотека небольших программ оболочки:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		sunmscapi.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		ucrtbase.dll:Microsoft® C Runtime Library:10.0.19041.789 (WinBuild.160101.0800):Microsoft Corporation
		vcruntime140_1.dll:Microsoft® C Runtime Library:14.29.30139.0 built by: vcwrkspc:Microsoft Corporation
		verify.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		win32u.dll:Win32u:10.0.19041.2604 (WinBuild.160101.0800):Microsoft Corporation
		windows.storage.dll:API хранения Microsoft WinRT:10.0.19041.2845 (WinBuild.160101.0800):Microsoft Corporation
		winrnr.dll:LDAP RnR Provider DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		wshbth.dll:Windows Sockets Helper DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		zip.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
Stacktrace:
	at net.minecraft.client.main.Main.method_44604(Main.java:205)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- System Details --
Details:
	Minecraft Version: 1.19.3
	Minecraft Version ID: 1.19.3
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 770543088 bytes (734 MiB) / 1711276032 bytes (1632 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 12
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 5 3600 6-Core Processor              
	Identifier: AuthenticAMD Family 23 Model 113 Stepping 0
	Microarchitecture: Zen 2
	Frequency (GHz): 3.59
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: NVIDIA GeForce GTX 1660
	Graphics card #0 vendor: NVIDIA (0x10de)
	Graphics card #0 VRAM (MB): 4095.00
	Graphics card #0 deviceId: 0x2184
	Graphics card #0 versionInfo: DriverVersion=31.0.15.1694
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 2.40
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MB): 8192.00
	Memory slot #1 clockSpeed (GHz): 2.40
	Memory slot #1 type: DDR4
	Virtual memory max (MB): 31180.39
	Virtual memory used (MB): 7904.94
	Swap memory total (MB): 14848.00
	Swap memory used (MB): 0.00
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		fabric-api: Fabric API 0.76.1+1.19.3
			fabric-api-base: Fabric API Base 0.4.21+70be179c85
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.20+49abcf7e85
			fabric-biome-api-v1: Fabric Biome API (v1) 12.1.2+f1c68e5885
			fabric-block-api-v1: Fabric Block API (v1) 1.0.5+e022e5d185
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.31+928d44d585
			fabric-client-tags-api-v1: Fabric Client Tags 1.0.12+1134c5b885
			fabric-command-api-v1: Fabric Command API (v1) 1.2.23+f71b366f85
			fabric-command-api-v2: Fabric Command API (v2) 2.2.2+2961b4a085
			fabric-commands-v0: Fabric Commands (v0) 0.2.40+df3654b385
			fabric-containers-v0: Fabric Containers (v0) 0.1.47+df3654b385
			fabric-content-registries-v0: Fabric Content Registries (v0) 3.5.3+1d37d50285
			fabric-convention-tags-v1: Fabric Convention Tags 1.3.0+521b5c5285
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.14+aeb40ebe85
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 11.3.3+40dc7a9485
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.42+00a2eb1085
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.5.7+b83334a085
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.40+3baeb27a85
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.45+df3654b385
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.30+99f9db8085
			fabric-item-api-v1: Fabric Item API (v1) 2.1.11+71b8257385
			fabric-item-group-api-v1: Fabric Item Group API (v1) 2.1.14+b5470ac785
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.31+bc01e09785
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.29+df3654b385
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.2.10+23a79c8a85
			fabric-loot-api-v2: Fabric Loot API (v2) 1.1.22+75e9821185
			fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.1.26+9e7660c685
			fabric-message-api-v1: Fabric Message API (v1) 5.1.0+c85585f885
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.33+49abcf7e85
			fabric-models-v0: Fabric Models (v0) 0.3.27+11ba9c3b85
			fabric-networking-api-v1: Fabric Networking API (v1) 1.2.17+4017a8cb85
			fabric-networking-v0: Fabric Networking (v0) 0.3.34+df3654b385
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 5.4.3+eb2a3ba985
			fabric-particles-v1: Fabric Particles (v1) 1.0.20+f1e4495b85
			fabric-recipe-api-v1: Fabric Recipe API (v1) 1.0.1+5176f73d85
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 2.1.0+f7b4d36485
			fabric-renderer-api-v1: Fabric Renderer API (v1) 2.2.2+3a95925a85
			fabric-renderer-indigo: Fabric Renderer - Indigo 0.8.0+3a95925a85
			fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.31+df3654b385
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.25+afca2f3e85
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.0.18+f1e4495b85
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.34+df3654b385
			fabric-rendering-v1: Fabric Rendering (v1) 1.13.0+ad17775585
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.3.0+e63306e085
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.11.1+b93ffe4d85
			fabric-screen-api-v1: Fabric Screen API (v1) 1.0.41+f1e4495b85
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.13+99f9db8085
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.8+75e9821185
			fabric-transfer-api-v1: Fabric Transfer API (v1) 2.1.15+ccd377ba85
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 2.3.1+40dc7a9485
		fabricloader: Fabric Loader 0.14.19
		java: OpenJDK 64-Bit Server VM 17
		minecraft: Minecraft 1.19.3
		optifabric: OptiFabric 1.13.24
			mm: Manningham Mills 2.3
		plasmovoice: Plasmo Voice 2.0.3
			aopalliance_aopalliance: aopalliance 1.0
			com_google_inject_guice: guice 5.0.1
			fabric-language-kotlin: Fabric Language Kotlin 1.9.1+kotlin.1.8.10
				org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.8.10
				org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.8.10
				org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.8.10
				org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.8.10
				org_jetbrains_kotlinx_atomicfu-jvm: atomicfu-jvm 0.19.0
				org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.6.4
				org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.6.4
				org_jetbrains_kotlinx_kotlinx-datetime-jvm: kotlinx-datetime-jvm 0.4.0
				org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.4.1
				org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.4.1
				org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.4.1
			fabric-permissions-api-v0: fabric-permissions-api 0.2-SNAPSHOT
			javax_inject_javax_inject: javax.inject 1
			universalcraft: UniversalCraft 262
			ustats: uStats 1.0.0
	Launched Version: fabric-loader-0.14.19-1.19.3
	Backend library: LWJGL version 3.3.1 SNAPSHOT
	Backend API: Unknown
	Window size: <not initialized>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	CPU: <unknown>
	OptiFine Version: OptiFine_1.19.3_HD_U_I3
	OptiFine Build: 20230324-184409
	Shaders: null
	OpenGlVersion: null
	OpenGlRenderer: null
	OpenGlVendor: null
	CpuCount: 0

-- OptiFabric --
Details:
	OptiFine jar designed for: 1.19.3
	OptiFine jar version: OptiFine_1.19.3_HD_U_I3
	OptiFine jar status: Valid OptiFine installer
	OptiFine remapped jar: C:/Users/pavsu/AppData/Roaming/.minecraft/.optifine/OptiFine_1.19.3_HD_U_I3/Optifine-mapped.jar
	OptiFabric error: <None>

**Comment by warjort** [comment]
`Caused by: java.lang.NullPointerException: Cannot invoke "su.plo.voice.client.config.VoiceClientConfig.setConfigFile(java.io.File)" because "this.config" is null
at su.plo.voice.client.BaseVoiceClient.loadConfig(BaseVoiceClient.java:290)`

Issue with the plasmo voice mod. 
Something to do with the configuration file.
Contact the mod author.
